### DISCUSSION TITLE
Can you run a relaxation through MACE?

### DISCUSSION THREAD

**Discussion by ceir13** [ACEsuit/mace#530] [Q&A] [answered]
Is it possible to run a POSCAR file through MACE to get the relaxed geometry of the structure and its energy? Is there any example of MACE relaxing a POSCAR, or could someone explain how to do it? Any help would be appreciated!

**Comment by wcwitt** [comment]
You can use ASE (https://wiki.fysik.dtu.dk/ase/index.html) to load the POSCAR and perform the relaxation. Use of the MACE ASE calculator is described here: https://mace-docs.readthedocs.io/en/latest/guide/ase.html.

**Comment by gabor1** [comment]
you need to convert your poscar to extended xyz using ASE. (ASE can read poscars and export extended xyz). When you say "run in it through mace", it's not really clear what you mean. mace is an ml force field architecture, in order to relax anything, one needs a trained ml model. there are models that are applicable to all materials (the mp0 series of models), but whether that pre-trained model is sufficient for your purposes (in terms of accuracy) depends on the application. normally one would fine-tune the model to get more accuracy.

**Reply by ceir13** [reply]
I am using the most recent pre-trained model from the MACE-MP-0 repository ([https://github.com/ACEsuit/mace-mp](url)). 

When I'm running the model with the XYZ files, are they single-point calculations or are they relaxing at multiple steps, and if so how do I view those multiple steps?

```python

``` calculator = MACECalculator(
    model_paths="/Users/ericjackson/Desktop/Lab/ML/MACE-MP-0/models/MACE_2024.Model",
    device="CPU",
)

```python

``` for filename in filenames:
    atoms = read(filename=f"{directory_path}/{filename}", format="vasp")
    atoms.calc = calculator
    energy = atoms.get_potential_energy()
    energy_list.append(energy)
print(energy_list)

**Reply by gabor1** [reply, marked as answer]
this is an ASE question. a get_potential_energy() call is exactly that - just gives you a potential energy. browse the ASE documentation on relaxations please. (I recommend PreconLBFGS)

**Reply by gabor1** [reply]
btw this is the latest model in that series: https://github.com/ACEsuit/mace-mp/releases/tag/mace_mp_0b

**Comment by gabor1** [comment]
you need to convert your poscar to extended xyz using ASE. (ASE can read poscars and export extended xyz). When you say "run in it through mace", it's not really clear what you mean. mace is an ml force field architecture, in order to relax anything, one needs a trained ml model. there are models that are applicable to all materials (the mp0 series of models), but whether that pre-trained model is sufficient for your purposes (in terms of accuracy) depends on the application. normally one would fine-tune the model to get more accuracy.

**Comment by bernstei** [comment]
> you need to convert your poscar to extended xyz using ASE.

ASE (`ase.io.read`) can read POSCARs directly, as long as their filename is "POSCAR" or ends has the suffix ".vasp" or ".poscar" (case insensitive, I believe)
